src/http/server.rs

```diff
@@ -159,15 +161,23 @@ impl From<&Configuration> for Newtype<httpd_ssl_config_t> {
         Self(httpd_ssl_config_t {
             httpd: http_config.0,
             session_tickets: false,
+            #[cfg(esp_idf_version_major = "5")]
```

Comment:
Please do `not(esp_idf_version_major = "4")`. This way we are safe for ESP-IDF 6 :)